federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -485,7 +463,6 @@ func (frsc *ReplicaSetController) reconcileReplicaSet(key string) (reconciliatio
 	if err != nil {
 		glog.Errorf("Failed to ensure delete object from underlying clusters finalizer in replicaset %s: %v",
 			frs.Name, err)
-		frsc.deliverReplicaSetByKey(key, 0, false)
```

Comment:
Also remove the call to deliverReplicaSetByKey in worker() when err != nil. In sync controller, it is called based on statusError.